### DISCUSSION TITLE
Future of MT ?

### DISCUSSION THREAD

**Discussion by wolfgang-ch** [mytourbook/mytourbook#914] [General]
I just learned in this video https://www.youtube.com/watch?v=ND86V8iglmE how to implement the bloom effect and I have an idea where to use it, then I saw this video https://www.youtube.com/watch?v=SQo9pTQ14Jk but this would be too much for MT :-)

**Comment by FJBDev** [comment]
>  and I have an idea where to use it

Where would you use it in MT ?

**Reply by wolfgang-ch** [reply]
> > and I have an idea where to use it
> 
> Where would you use it in MT ?

This is top secrect :-)

In the last few day's I searched heavily for this glow/bloom effect, e.g.

https://inspirnathan.com/posts/65-glow-shader-in-shadertoy/
https://www.shadertoy.com/view/4dfGzs
https://learnopengl.com/Advanced-Lighting/Bloom

but the video which I found now, seems to help me what I searched for

**Reply by wolfgang-ch** [reply]
> > > and I have an idea where to use it
> > 
> > 
> > Where would you use it in MT ?
> 
> This is top secrect :-)

This https://github.com/mytourbook/mytourbook/issues/516#issuecomment-1226853304 may answer the question

**Reply by FJBDev** [reply]
I would love to know about the future of MT 😉

**Reply by wolfgang-ch** [reply]
There is no master plan for MT, the implemented features are mostly spontaneous but I'm also looking on the feature requests/bugs and be implementing/fixing these sometimes.

Currently my focus in on the 2.5D map because there are so many missing things compared with 2D map, but for me this is very time consuming because of the maths you have to understand. There are many OpenGL tutorials you can learn from but this is also time consuming. 

The tutorials which were linked in https://github.com/mytourbook/mytourbook/issues/516#issuecomment-1227615879 were very interresting but the author was talking so fast, to be able to understand all immediately, what he explained.

**Reply by wolfgang-ch** [reply]
> > > and I have an idea where to use it
> > 
> > Where would you use it in MT ?
> 
> This is top secrect :-)

This was the initial idea to "glow" the directory arrow but now I'm a little bit disappointed that it did't look like expected:

* Arrows are not smoothly moved
* All not visible arrows are also displayed however with opacity 0 and are partly covering the moving arrow

-> there needs some improvements :-)

https://user-images.githubusercontent.com/1283445/190404801-119ece31-7317-47df-8e84-0f3203a73c67.mp4

**Reply by FJBDev** [reply]
One feature that I think would be nice to have (not really "useful" but nice/cool) would be the live drawing of the tour such as this [Relive.cc](https://relive.cc) example
https://www.youtube.com/watch?v=SazTPv9_x0c

I even asked for help in the [WW forum](https://forum.worldwindcentral.com/forum/world-wind-java-forums/development-help/160528-draw-a-gps-track-slowly) as I was wondering if it's possible to do in the 3D map but no answers yet....

**Reply by FJBDev** [reply]
> -> there needs some improvements :-)
> m25-moving-arrows2.mp4


Maybe there needs to be some improvements but it still looks great 👍 
That reminds me of the famous ["DotRacing" plugin](https://www.youtube.com/watch?v=y6qglL2Ps_A) of the old Sporttracks where one could compare several activities to see with little animated people who was "winning" the race. It was useful, for example, to compare several similar activities and see where time was gained or lost etc.... I could see being useful in MT

**Reply by wolfgang-ch** [reply]
> One feature that I think would be nice to have (not really "useful" but nice/cool) would be the live drawing 

Done :-)

This was a 2 minute hack https://github.com/mytourbook/mytourbook/commit/918424fea05e4aeb6d410d5f18be0dd1f8efeef9

https://user-images.githubusercontent.com/1283445/190433879-0185ee45-0ae0-4a8c-8f12-bac60c2e7324.mp4

**Reply by FJBDev** [reply]
> Done :-)
> 
> This was a 2 minute hack [918424f](https://github.com/mytourbook/mytourbook/commit/918424fea05e4aeb6d410d5f18be0dd1f8efeef9)

WOW!!!! Can you now do that for the 2D Map and 3D Map for the next release ?? :blush:

**Reply by wolfgang-ch** [reply]
This hack works only when the whole tour is displayed, otherwise it is much more complex.

For the 3D map, I think this should work in the similar way but for the 2D map it total different and for every progress the map needs to be repainted which will likely have a performance problem.
